sklearn_numba_dpex/utils/tests/test_device_params.py

```diff
@@ -0,0 +1,55 @@
+import pytest
+import warnings
+
+import dpctl
+
+from sklearn_numba_dpex.utils._device import _DeviceParams
+
+
+def test_opencl_requirement():
+    device = dpctl.select_default_device()
+    has_preferred_work_group_size_multiple = hasattr(
+        device, "preferred_work_group_size_multiple"
+    )
+    has_global_mem_cache_size = hasattr(device, "global_mem_cache_size")
+    assert not (
+        has_preferred_work_group_size_multiple or has_global_mem_cache_size
+    ), "pyopencl is not required anymore, use dpctl instead"
+
+
+class _FakeSyclDevice:
+    def __init__(
+        self,
+        has_aspect_fp16=True,
+        has_aspect_fp64=True,
+        max_work_group_size=16,
+        name="Fake Sycl Device",
+    ):
+        self.has_aspect_fp16 = has_aspect_fp16
+        self.has_aspect_fp64 = has_aspect_fp64
+        self.max_work_group_size = 16
```

Comment:
What about using a [`dataclass`](https://docs.python.org/3/library/dataclasses.html) for `_FakeSyclDevice`?
```suggestion
        self.max_work_group_size = max_work_group_size
```

Reply:
ok